spark/src/test/scala/org/apache/spark/sql/delta/DeltaVacuumSuite.scala

```diff
@@ -578,6 +586,230 @@ class DeltaVacuumSuite
     }
   }
 
+  test("schema validation for vacuum by using inventory dataframe") {
+    withEnvironment { (tempDir, clock) =>
+      val deltaLog = DeltaLog.forTable(spark, tempDir, clock)
+      val txn = deltaLog.startTransaction()
+      val schema = new StructType().add("_underscore_col_", IntegerType).add("n", IntegerType)
+      val metadata =
+        Metadata(schemaString = schema.json, partitionColumns = Seq("_underscore_col_"))
+      txn.commit(metadata :: Nil, DeltaOperations.CreateTable(metadata, isManaged = true))
+
+      val expectedInventorySchema = StructType(
```

Comment:
It might be worth pulling this out into the parent class, because it's reused in a lot of tests now.